pkg/status/controller_test.go

```diff
@@ -0,0 +1,346 @@
+							Reason:             "",
+							Message:            "",
+						},
+						{
+							Type:               configv1.OperatorDegraded,
+							Status:             configv1.ConditionTrue,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "",
+							Message:            "",
+						},
+						{
+							Type:               configv1.OperatorProgressing,
+							Status:             configv1.ConditionTrue,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "",
+							Message:            "",
+						},
+						{
+							Type:               configv1.OperatorUpgradeable,
+							Status:             configv1.ConditionFalse,
+							LastTransitionTime: metav1.Now(),
+							Reason:             "",
+							Message:            "",
+						},
+					},
+					Versions: []configv1.OperandVersion{
+						{
+							Name:    "anything",
+							Version: "anything",
+						},
+					},
+					RelatedObjects: []configv1.ObjectReference{
+						{
+							Group:    "",
+							Resource: "anything",
+							Name:     "anything",
+						},
+					},
+				},
+			},
+		}),
+	)
+})
+
+var _ = Describe("Filters are exclusive", func() {
+	It("Should allow only 'machine-approver' clusterOperators", func() {
+		operator := &configv1.ClusterOperator{}
+		operator.SetName(clusterOperatorName)
+		Expect(clusterOperatorFilter(operator)).To(BeTrue())
+		operator.SetName("unknown")
+		Expect(clusterOperatorFilter(operator)).To(BeFalse())
+		Expect(clusterOperatorFilter(nil)).To(BeFalse())
+	})
+
+	It("Should allow only CSR objects", func() {
+		Expect(csrFilter(&certificatesv1.CertificateSigningRequest{})).To(BeTrue())
+		Expect(csrFilter(nil)).To(BeFalse())
+	})
+})
+
+var _ = Describe("toClusterOperator mapping is targeting requests to 'machine-approver' clusterOperator", func() {
+	It("Should map reconciles to 'machine-approver' CO", func() {
+		object := certificatesv1.CertificateSigningRequest{}
+		mapObject := handler.MapObject{
+			Meta:   object.GetObjectMeta(),
+			Object: &object,
+		}
+		requests := []reconcile.Request{{
+			NamespacedName: client.ObjectKey{
+				Name: clusterOperatorName,
+			},
+		}}
+		Expect(toClusterOperator(mapObject)).To(Equal(requests))
+	})
+})
+
+var _ = Describe("Reconcile triggers", func() {
+	var operator *configv1.ClusterOperator
+	var csr *certificatesv1.CertificateSigningRequest
+	var env *envtest.Environment
+	var c client.Client
+	var done chan struct{}
+	var reconciled int
+	var reconciledTimes = func() int { return reconciled }
+
+	BeforeEach(func() {
+		reconciled = 0
+		done = make(chan struct{})
+		operator = &configv1.ClusterOperator{}
+		operator.SetName(clusterOperatorName)
+
+		csr = &certificatesv1.CertificateSigningRequest{
+			Spec: certificatesv1.CertificateSigningRequestSpec{
+				Request: []byte(csrData),
+			},
+		}
+		csr.SetName("csrtest")
+
+		By("Setting up test environment")
+		env = &envtest.Environment{
+			CRDDirectoryPaths: []string{filepath.Join("clusteroperator.crd.yaml")},
+		}
```

Comment:
Similar concerns, this will be very slow

Reply:
In comparison to the default openshift api, I was not able to find an importable location with needed `CRDs`, and don't like an idea to put this into code at all. Am I missing something?